### ISSUE TITLE
Difficulty installing / http 403 access denied

### ISSUE THREAD

**Issue by markb81** [kimai/kimai#774]
Hi,

I'm experiencing some difficulty installing kimai in my hosting space with my hosting provider.
I copied all files, created a database etc. I run the /installer. All checks succeed. I finish the installation. The last step is the admin/changme message. Then proceed and the installation fails.

Internet explorer gives http/403 saying access denied. Google Chrome just presents a blank page. I then chmodded my complete directory to 777. I reset the MysQL password to something simple without chars. But nothing seems to do the trick.

Hope somebody can help me cause I would really like to use the software.

Thanks!

Mark

**Comment by kevinpapst** [comment]
Hi Mark, unlikely that this is cause by Kimai ... but you never know.
So you can open http://yourDomain/kimai/installer/index.php ?
Can you access http://yourDomain/kimai/index.php ?
Do you have a Webserver log, that might provide more info?
Check the logfile at kimai/temporary/logfile.txt

**Comment by markb81** [comment]
Hi Kevin,

Thanks for your quick reply. I can open the installer and follow all steps. Selecting database etc etc. If I open index.php it redirects me to installer. I don't have access to webserver logs, but I can ask my ISP to provide them. Which one would be good to inspect? The /temporary/logfile.txt doesn't exist. The temporary directory exists but is empty.

I just upgraded my webspace to PHP7. Is PHP7 supported?

Kind regards,

Mark

**Comment by kevinpapst** [comment]
If you get redirected to the installer, there were problems within the installation.
Was the file includes/autoconf.php generated?

**Comment by markb81** [comment]
Hi,

The autoconf.php is created and filled with the data I supplied during the setup.

Kind regards,

Mark

**Comment by markb81** [comment]
Hi,

I redid the installation (just wiped the dir & databases etc). no luck. I set PHP version back to 5.6 but that doesn't seem to change a thing.

Is there  a way to generate some debuglogging from Kimai?

All setup options are green. Autoconfig is created. The database remains empty.

Kind regards,

Mark

**Comment by markb81** [comment]
It fails on this url: http://kimai.mydomain.nl/installer/install.php?accept=1&timezone=Europe/Amsterdam - just a blank screen in Chrome or 403 error in IE.

**Comment by kevinpapst** [comment]
Now its getting really weird. First post you said the installation finishes correctly, now a blank screen???
When you said the database remains empty, does that mean no tables were created or there is no data within the tables? In first case, that could be a problem, that your database user has not sufficient permissions to create tables.

**Comment by markb81** [comment]
Hi Kevin,

Well the installation does finish. All the installation steps are okay-ed and after the installation it redirects to the url I mentioned in the comment above.

Google Chrome presents a blank page with this URL. Internet Explorer does 403 access denied. I checked the database but it's completely empty. The database user has all permissions except grant.

The autoconf.php is configured. The temporary directory remains empty. So no logs in there.

I tried on PHP7 & PHP 5.6 . 

Kind regards,

Mark

**Comment by kevinpapst** [comment]
Hey Mark, 
that really sounds like a database problem. Especially when you tried two PHP versions and the installation works fine on several systems.

1. Do you use a pre-existing database or are you telling Kimai to create a database?

2. Please login using PHPmyadmin (or whatever tool you use) with the user that you provide during the installation and execute some SQL statements from here (replace ${p} with "kimai_"):
https://github.com/kimai/kimai/blob/master/installer/install.php

For example
```
CREATE TABLE `kimai_users` (
  `userID` int(10) NOT NULL AUTO_INCREMENT PRIMARY KEY,
  `name` varchar(160) NOT NULL,
  `alias` varchar(160),
  `trash` tinyint(1) NOT NULL default '0',
  `active` tinyint(1) NOT NULL default '1',
  `mail` varchar(160) NOT NULL DEFAULT '',
  `password` varchar(254) NULL DEFAULT NULL,
  `passwordResetHash` char(32) NULL DEFAULT NULL,
  `ban` int(1) NOT NULL default '0',
  `banTime` int(10) NOT NULL default '0',
  `secure` varchar(60) NOT NULL default '0',
  `lastProject` int(10) NOT NULL default '1',
  `lastActivity` int(10) NOT NULL default '1',
  `lastRecord` int(10) NOT NULL default '0',
  `timeframeBegin` varchar(60) NOT NULL default '0',
  `timeframeEnd` varchar(60) NOT NULL default '0',
  `apikey` varchar(30) NULL DEFAULT NULL,
  `globalRoleID` int(10) NOT NULL,
  UNIQUE KEY `name` (`name`),
  UNIQUE KEY `apikey` (`apikey`)
);
```

**Comment by markb81** [comment]
Hi Kevin,

I got this to work. I tried to install kimai in a sub domain. However when creating a subdomain it just creates a folder in my root domain hostingspace and links it to the sub domain.

My root domain runs Wordpress with iThemes Security which rewrote the .htaccess in the root domain. I disabled iThemes, installed kimai and re-enabled iThemes. Kimai is now running smooth. 

So you were right. It wasn't Kimai :-) Thanks for thinking with me.
